addon/services/file-manager.js

```diff
@@ -245,6 +243,64 @@ export default Ember.Service.extend({
     },
 
     /**
+     * Check whether the given url corresponds to a model that is currently
+     * reloading after a file operation.
+     *
+     * Used by `mixin:file-cache-bypass` to avoid a race condition where the
+     * cache might return stale, inaccurate data.
+     *
+     * @method isReloadingUrl
+     * @param {String} url
+     * @return {Boolean} `true` if `url` corresponds to a pending reload on a
+     * model immediately after a Waterbutler action, otherwise `false`.
+     */
+    isReloadingUrl(url) {
+        return !!this._reloadingUrls[url];
+    },
+
+    /**
+     * Hash set of URLs for `model.reload()` calls that are still pending.
+     *
+     * @property _reloadingUrls
+     * @private
+     */
+    _reloadingUrls: {},
+
+    /**
+     * Force-reload a model from the API.
+     *
+     * @method _reloadModel
+     * @private
+     * @param {Object} model `file` model or a `files` relationship
+     * @return {Promise} Promise that resolves to the reloaded model or
+     * rejects with an error message.
+     */
+    _reloadModel(model) {
+        // If it's a file model, it has its own URL in `links.info`.
+        let reloadUrl = model.get('links.info');
+        if (!reloadUrl) {
```

Comment:
Purely for future reference, this can be written more concisely using lazy evaluation, as:

`let reloadUrl = model.get('links.info') || model.get('content.relationship.link');`

This sort of pattern is particularly common in JS, where (transpiling aside) people are trying to write more concise code to save bandwidth. Fortunately, in this case, it's not a big difference in readability. :)